Restructure docs, stage 1

patiencedaur opened the pull request:
Part of this epic: https://github.com/tarantool/doc/issues/2051
Part of https://github.com/tarantool/doc/issues/2805
Resolves https://github.com/tarantool/doc/issues/2487
Resolves https://github.com/tarantool/doc/issues/1467

Deployment: https://docs.d.tarantool.io/ru/doc/restructure-docs-1/

- Releases
- [x] Rename Release notes to Releases
- [x] Add a link to Downloads to the Releases page
- [x] Structure release notes by series (2.10, 1.10, EOL versions)
- [x] Replace links accordingly
- [x] Remove RN of outdated prerelease versions

- Overview
- [x] Write the overview article "What is Tarantool?"
- [x] Add the Overview section

- How-to guides
- [x] Add the How-to guides section
- [x] Move "What's next?" tutorials from Getting started to the new section
- [x] Remove the old Tutorials section and put all the tutorials into How-to guides
- [x] Add more tutorials as per https://www.notion.so/tarantool/Tarantool-CE-c075d0db9ce24893beca704c145ae501#704bee0eff07402fa4db192ef76923d8


See https://www.notion.so/tarantool/Tarantool-CE-c075d0db9ce24893beca704c145ae501#fe407fc7c3184d50b0bef804c760b719.

patiencedaur committed:
Update the Releases section

patiencedaur committed:
Update the Releases section

patiencedaur committed:
Update the Releases section

patiencedaur committed:
Clarify link between number and cdata (#3050)

* Clarify the connection between number and cdata
* Adjust table widths
* Mention decimal among number field index types

Resolves #3049
Resolves #3040

patiencedaur committed:
Remove redundant text on index page

patiencedaur committed:
Add a note about recipes in tarantool/examples (#3055)

Resolves #1344

patiencedaur committed:
Add the Overview article draft

patiencedaur committed:
Attempt to fix panels

Panels are not available for preview in the local build.

* Add Overview to the index

* Fix panel formatting

NickVolynkin commented on locale/ru/LC_MESSAGES/reference/reference_lua/box_index/pairs.po:
```diff
@@ -34,14 +34,15 @@ msgstr ""
 "все кортежи в порядке по возрастанию или по убыванию, начиная с указанного "
 "ключа. Однако другие типы индексов не поддерживают упорядочивание."
 
+#, fuzzy
```
This update breaks translations in a lot of places. For example, page https://docs.d.tarantool.io/ru/doc/restructure-docs-1/reference/reference_lua/box_index/pairs/

patiencedaur replied:
Looks like make update-po worked in an unexpected way. I can just remove this commit. In this case, what is the correct way to batch update translations?

patiencedaur committed:
Apply suggestions from initial review

p7nov committed:
Add tt check, logrotate, connect (#3051)

Resolves #2557
Resolves #2558
Resolves #2559

veod32 committed:
Add a new section for the datetime module with the description of its main functions and methods

Closes #2411, #2576

veod32 committed:
Correct description of applying data schema on the Code tab in the schema.yml file (#3062)

Fixes #3053

patiencedaur committed:
Remove panels

patiencedaur commented:
Redirect after this PR:

- https://www.tarantool.io/en/doc/latest/tutorials/ -> https://www.tarantool.io/en/doc/latest/howto/
- https://www.tarantool.io/ru/doc/latest/tutorials/ -> https://www.tarantool.io/ru/doc/latest/howto/

.

- https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_space/examples/ -> https://www.tarantool.io/en/doc/latest/how-to/crud
- https://www.tarantool.io/ru/doc/latest/reference/reference_lua/box_space/examples/ -> https://www.tarantool.io/ru/doc/latest/how-to/crud

.

- https://www.tarantool.io/en/doc/latest/getting_started/getting_started_db/ -> https://www.tarantool.io/en/doc/latest/how-to/getting_started_db/
- https://www.tarantool.io/ru/doc/latest/getting_started/getting_started_db/ -> https://www.tarantool.io/ru/doc/latest/how-to/getting_started_db/

.

- https://www.tarantool.io/en/doc/latest/getting_started/getting_started_connectors/ -> https://www.tarantool.io/en/doc/latest/how-to/getting_started_connectors/
- https://www.tarantool.io/ru/doc/latest/getting_started/getting_started_connectors/ -> https://www.tarantool.io/ru/doc/latest/how-to/getting_started_connectors/

.

- https://www.tarantool.io/en/doc/latest/getting_started/getting_started_python/ -> https://www.tarantool.io/en/doc/latest/how-to/getting_started_python/
- https://www.tarantool.io/ru/doc/latest/getting_started/getting_started_python/ -> https://www.tarantool.io/ru/doc/latest/how-to/getting_started_python/

.

- https://www.tarantool.io/en/doc/latest/getting_started/getting_started_php/ -> https://www.tarantool.io/en/doc/latest/how-to/getting_started_php/
- https://www.tarantool.io/ru/doc/latest/getting_started/getting_started_php/ -> https://www.tarantool.io/ru/doc/latest/how-to/getting_started_php/

.

- https://www.tarantool.io/en/doc/latest/getting_started/getting_started_go/ -> https://www.tarantool.io/en/doc/latest/how-to/getting_started_go/
- https://www.tarantool.io/ru/doc/latest/getting_started/getting_started_go/ -> https://www.tarantool.io/ru/doc/latest/how-to/getting_started_go/

.

- https://www.tarantool.io/en/doc/latest/getting_started/getting_started_cxx/ -> https://www.tarantool.io/en/doc/latest/how-to/getting_started_cxx/
- https://www.tarantool.io/ru/doc/latest/getting_started/getting_started_cxx/ -> https://www.tarantool.io/ru/doc/latest/how-to/getting_started_cxx/

.

- https://www.tarantool.io/en/doc/latest/getting_started/getting_started_cartridge/ -> https://www.tarantool.io/en/doc/latest/how-to/getting_started_cartridge/
- https://www.tarantool.io/en/doc/latest/getting_started/getting_started_cartridge/ -> https://www.tarantool.io/ru/doc/latest/how-to/getting_started_cartridge/

.

- https://www.tarantool.io/en/doc/latest/reference/reference_sql/sql_beginners_guide/ ->  https://www.tarantool.io/en/doc/latest/how-to/sql/sql_beginners_guide
- https://www.tarantool.io/ru/doc/latest/reference/reference_sql/sql_beginners_guide/ -> https://www.tarantool.io/ru/doc/latest/how-to/sql/sql_beginners_guide

.

- https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_space/examples/ -> https://www.tarantool.io/en/doc/latest/how-to/crud
- https://www.tarantool.io/ru/doc/latest/reference/reference_lua/box_space/examples/ -> https://www.tarantool.io/ru/doc/latest/how-to/crud

patiencedaur committed:
Create the new How-to section

patiencedaur committed:
Fix EOL date for 2.8.2 (#3066)

Resolves #3064

patiencedaur committed:
Create the new How-to section structure

patiencedaur committed:
Move 'Getting started/What's next' to the new section

patiencedaur committed:
Fix some links

patiencedaur committed:
Move Tarantool CXX connector guide

xuniq committed:
Document the watchers and built-in events (#2858)

Fixes #2551
Fixes #2409
Fixes #2407

* Add new functions to module box
* Update watch and broadcast methods
* Add System events section
* Add glossary
* Proofread text and add links

Written by Kseniia Antonova
Reviewed by Vladimir Davydov and Dmitry Oboukhov
Proofread by Patience Daur

Co-authored-by: Kseniia Antonova <xuniq.is.here@gmail.com>
Co-authored-by: Patience Daur <patiencedaur@gmail.com>

patiencedaur committed:
Fix build warnings (#3068)

xuniq committed:
Document IPROTO watchers (#3044)

Written by Kseniia Antonova
Proofread by Patience Daur

Co-authored-by: Kseniia Antonova <xuniq.is.here@gmail.com>
Co-authored-by: Patience Daur <patiencedaur@gmail.com>

patiencedaur committed:
Fix small issues (#3069)

* Provide links to Lua and LuaJIT data types

Resolves #1318

* Replace wrong link to Lua 5.2 with correct to Lua 5.1

Resolves #1321

* Provide a more relevant link to Lua table manual

Resolves #1785

* Add msgpack.NULL to data types table

Resolves #927

* Add a sharded space example

Resolves #2474

* Clarify box.cfg.listen value type

Resolves #2855

patiencedaur committed:
Add SQL Beginners' Guide to How-to

patiencedaur committed:
Add CRUD operations how-to guide

* Add guide
* Fix links

xuniq committed:
Add default timeout in module http (#3076)

Fixes #3074 

* Add timeout
* Fix formatting
* Fix dashes
* Update .po files

xuniq committed:
Fix typo in box.watch example (#3078)

p7nov committed:
Add 2.10.1 release notes (#3073)

Resolves #3070
Co-authored-by: Kseniia Antonova <73473519+xuniq@users.noreply.github.com>

p7nov committed:
Add 1.10.14 LTS release notes (#3075)

Resolves #3071

Co-authored-by: Kseniia Antonova <73473519+xuniq@users.noreply.github.com>

patiencedaur committed:
Add missing quotes in an example (#3081)

NickVolynkin committed:
Use strict error handling in bash scripts (#3083)

NickVolynkin committed:
Use strict error handling in bash scripts (#3083)

p7nov committed:
Restore links to new RNs

p7nov committed:
Merge branch 'latest' into restructure-docs-1

p7nov committed:
Fix releases page TOC

p7nov committed:
Restructure the releases section

p7nov committed:
Remove irrelevant beta and rc release notes

p7nov commented:
More redirects:

* https://www.tarantool.io/en/doc/latest/release/2.10.0-rc1/ > https://www.tarantool.io/en/doc/latest/release/2.10.0/
* https://www.tarantool.io/en/doc/latest/release/2.10.0-beta1/ > https://www.tarantool.io/en/doc/latest/release/2.10.0/
* https://www.tarantool.io/en/doc/latest/release/2.10.0-beta2/ > https://www.tarantool.io/en/doc/latest/release/2.10.0/

patiencedaur commented:
How to make redirects (internal document): https://www.notion.so/tarantool/5b0e1ab45d0841a3b502cfea89829ef7

patiencedaur committed:
Point to the Cartridge tutorial explicitly

patiencedaur committed:
Merge branch 'latest' into restructure-docs-1

patiencedaur committed:
Move the How-to section up the menu

Product suggestion.
